drivers: net: ethernet: realtek: r8168: modify led status

RadxaSongjun opened the pull request:
drivers: net: ethernet: realtek: r8168: modify led status

Signed-off-by: SongJun Li <lisongjun@radxa.com>

RadxaYuntian commented on drivers/net/ethernet/realtek/r8168/r8168_n.c:
```diff
@@ -24002,6 +24004,11 @@ rtl8168_init_software_variable(struct net_device *dev)
 
         tp->NicCustLedValue = RTL_R16(tp, CustomLED);
 
+        if(tp->NicCustLedValue != RADXA_CUSTOM_LED) {
+                printk("status = 0x%x, RADXA_STATUS = 0x%x\r\n", tp->NicCustLedValue, RADXA_CUSTOM_LED);
+                RTL_W16(tp, CustomLED, RADXA_CUSTOM_LED);
+        }
+
```
```suggestion
        RTL_W16(tp, CustomLED, RADXA_CUSTOM_LED);
        tp->NicCustLedValue = RTL_R16(tp, CustomLED);
```
不过最好是添加一个设备树属性来override这个值吧

RadxaYuntian commented on drivers/net/ethernet/realtek/r8168/r8168_n.c:
```diff
@@ -23446,6 +23446,8 @@ rtl8168_get_bios_setting(struct net_device *dev)
         }
 }
 
+#define RADXA_CUSTOM_LED 0x0c7f
```
```suggestion
// Definition: https://blog.csdn.net/Yongheng6/article/details/134659910
#define RADXA_CUSTOM_LED 0x0c7f
```

RadxaSongjun replied:
if(tp->NicCustLedValue != RADXA_CUSTOM_LED)加入这个判断是因为sfuse不能烧入太多次而做的一个检测机制，设备树override我试试看

RadxaYuntian replied:
这个应该是写寄存器，不是efuse啊

RadxaStephen commented:
Please check this patch. https://github.com/coolsnowwolf/lede/blob/master/package/lean/r8168/patches/100-add-LED-configuration-from-OF.patch

RadxaSongjun committed:
drivers: net: ethernet: realtek: r8168: add LED configuration from OF

This patch is from https://github.com/coolsnowwolf/lede/blob/master/package/lean/r8168/patches/100-add-LED-configuration-from-OF.patch

Signed-off-by: SongJun Li <lisongjun@radxa.com>

RadxaSongjun committed:
arm64: dts: rockchip: e20c: add RTL8111H led configuration

set realtek,led-data = <0x007f> to Keep the network port connection status light always on, signal light blinking, and turn off unused led2

Signed-off-by: SongJun Li <lisongjun@radxa.com>

RadxaStephen commented:
I change the code like removing unnecessary space and change the git message.

Added to branch linux-5.10-gen-rkr4.1.
